src/config.rs

```diff
@@ -85,6 +86,23 @@ impl Config {
             }
         }
 
+        if let Some(config_file) = &opts.config_file {
+            // change the working directory to the place where the config file is, so that all
+            // paths are relative to the config file instead of the location where the templexer is called from
+            let base_path = config_file.parent().ok_or_else(|| {
+                anyhow!(
+                    "could not determine base path of given configuration file {:?}",
+                    config_file
+                )
+            })?;
+
+            // if we're already next to the config file, the base path will be empty
+            if !base_path.to_string_lossy().is_empty() {
```

Comment:
```suggestion
            if !base_path.as_os_str().is_empty() {
```

Is probably better than converting it to a utf8 string. The PathBuf is backed by OsStr so that would also mean no allocation.

https://github.com/rust-lang/rust/pull/31877#issuecomment-191901957

Maybe checking for the length of the components would be even better but I think as is should be fine.

Reply:
```suggestion
            if !base_path.components().next().is_none() {
```

Would be how checking the components would look like. I find it less clear what the intention is than the string/osstr approach but I guess its fine.

Reply:
That seems to be the more elegant way to do it 👍🏻 thanks

Reply:
And clippy is right again :)
![image](https://user-images.githubusercontent.com/188284/130575388-31bc91e2-cceb-4d02-9667-b78b3bfc2ed4.png)